### PR TITLE
Allow for multiple timestep limiting functions 

### LINKED ISSUE #17642: Multiple `timestep_limiting_function` in `IterationAdaptiveDT`
## Reason
In simulations of transients governed by multiple linear ramps (on mechanical BCs and thermal BCs for example) it would be useful to be able to specify multiple `timestep_limiting_function`s in `IterationAdaptiveDT`. That way the union of all nodes in all functions could be hit and the timestep could be limited by the maximum function value change in any of the listed functions.

## Design
Make the parameter accept a vector of FunctionNames, obtain the set union of all time nodes, use the minimum of all maximum function change timesteps.

## Impact
Added capability.

### PR DESCRIPTION
**Pull request by dschwen**
Closes #17642

### CODE REVIEW

File: framework/src/timesteppers/IterationAdaptiveDT.C [hunk 1]
```diff
@@ -318,79 +333,80 @@ void
 IterationAdaptiveDT::limitDTToPostprocessorValue(Real & limitedDT) const
 {
   if (_pps_value && _t_step > 1)
-  {
```
Comment (bwspenc) [inline framework/src/timesteppers/IterationAdaptiveDT.C:321]: I know that our style guide says to not have curly braces around one-liners, but in cases like this where there's a nested multi-line block, it seems to me like including the curly braces improves readability.

File: test/tests/time_steppers/iteration_adaptive/tests [hunk 2]
```diff
@@ -27,78 +27,92 @@
...
     requirement = "MOOSE shall support the ability to force time steps to resolve
     sudden changes in piecewise linear functions."
   [../]
 
+  [./multi_piecewise_linear_function_point]
+    type = 'CSVDiff'
+    input = 'multi_piecewise_linear.i'
+    csvdiff = 'multi_piecewise_linear_function_point.csv'
+    cli_args = "Executioner/TimeStepper/force_step_every_function_point=true Outputs/file_base=multi_piecewise_linear_function_point"
+    requirement = "MOOSE shall support the ability to force time steps to hit all nodes in a set of piecewide linear functions."
```
Comment (bwspenc) [inline test/tests/time_steppers/iteration_adaptive/tests:109]: piecewide->piecewise

**Review by bwspenc — CHANGES_REQUESTED** [review]
Looks good! Just one minor thing as noted. I'm fine either way on the curly braces, but I'd prefer to leave them as they were.

**Comment by dschwen** [comment]
Suddenly an unrelated failure in Sockeye pops up :-/

**Review by bwspenc — APPROVED** [review]

**Comment by tophmatthews** [comment]
I think somehow in here you cut out the ability to use `PiecewiseConstant` as a timestep limiting function @dschwen ? `PiecewiseLinear` works, `PiecewiseConstants` are accumulated, but not interrogated...

**Comment by dschwen** [comment]
Hm, I went over my changes, but I don't see anything to that effect. This PR keeps the same `dynamic_cast`s. Did `PiecewiseConstant` ever work?

**Comment by tophmatthews** [comment]
It may not be your changes...just that you're the last to change. Look at how `_piecewise_timestep_limiting_functions` is used vs `_piecewise_linear_timestep_limiting_functions`

**Comment by dschwen** [comment]
Maybe #14221 is the culprit @tophmatthews

**Pull request merged**